guacamole-ext/src/main/java/org/apache/guacamole/token/TokenFilter.java

```diff
@@ -273,6 +347,34 @@ public void filterValues(Map<?, String> map) {
             String value = entry.getValue();
             if (value != null)
                 entry.setValue(filter(value));
+
+        }
+
+    }
+
+    /**
+     * Given an arbitrary map containing String values, replace each non-null
+     * value with the corresponding filtered value. If any token in any string
+     * has no defined value within this TokenFilter, a
+     * GuacamoleTokenUndefinedException will be thrown.
+     *
+     * @param map
+     *     The map whose values should be filtered.
+     *
+     * @throws GuacamoleTokenUndefinedException
+     *     If at least one token in at least one string has no corresponding
+     *     value.
+     */
+    public void filterValuesStrict(Map<?, String> map)
```

Comment:
I think this API might be a little more straightforward if there were a 1 and 2 argument version each of `filterValue` and `filterValues`, with the 1-argument version passing `false` for the `strict` parameter. That would also reduce the duplicated iteration code in this function.

Reply:
(See above - this would end up requiring exception handling on the part of the developer even though it's known the exception would never be thrown).